active/radiobuttons/RadioButtons_Spec.md

```diff
@@ -169,84 +169,120 @@ If you find that you need wrapping behavior, perhaps a RadioButtons group isn't
 
 # API Details
 
-## RadioButtons
+## RadioButtons IDL
 
-```
-namespace Microsoft.UI.Xaml.Controls
+``` c++
+namespace MU_XC_NAMESPACE
 {
+
+[WUXC_VERSION_PREVIEW]
 [webhosthidden]
 [contentproperty("Items")]
+[MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+[MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]
 unsealed runtimeclass RadioButtons : Windows.UI.Xaml.Controls.Control
 {
     RadioButtons();
 
     Object ItemsSource;
     Windows.Foundation.Collections.IVector<Object> Items{ get; };
-    Windows.UI.Xaml.DataTemplate ItemTemplate;
+    [MUX_DEFAULT_VALUE("winrt::RadioButtonsElementFactory{}")]
+    Object ItemTemplate;
 
-    Windows.UI.Xaml.DependencyObject ContainerFromItem(Object item);
-    Windows.UI.Xaml.DependencyObject ContainerFromIndex(Int32 index);
+    Windows.UI.Xaml.UIElement ContainerFromIndex(Int32 index);
 
+    [MUX_DEFAULT_VALUE("-1")]
     Int32 SelectedIndex;
-    Object SelectedItem;
+    Object SelectedItem{ get; };
     event Windows.UI.Xaml.Controls.SelectionChangedEventHandler SelectionChanged;
 
-    Int32 MaxColumns;
+    [MUX_DEFAULT_VALUE("1")]
+    Int32 MaximumColumns;
     Object Header;
 
     static Windows.UI.Xaml.DependencyProperty ItemsSourceProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty ItemsProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty ItemTemplateProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty SelectedIndexProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty SelectedItemProperty{ get; };
-    static Windows.UI.Xaml.DependencyProperty MaxColumnsProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty MaximumColumnsProperty{ get; };
     static Windows.UI.Xaml.DependencyProperty HeaderProperty{ get; };
 }
-}
-```
-
-## RadioButtons Automation Peer
 
-```
-namespace Microsoft.UI.Xaml.Automation.Peers
-{
-[webhosthidden]
-unsealed runtimeclass RadioButtonsListViewItemAutomationPeer : Windows.UI.Xaml.Automation.Peers.ListViewItemAutomationPeer
-{
-    RadioButtonsListViewItemAutomationPeer(RadioButtonsListViewItem owner);
-}
 }
+
 ```
 
 ## RadioButtons Primitives
 
-```
-namespace Microsoft.UI.Xaml.Controls.Primitives
+``` c++
+namespace MU_XCP_NAMESPACE
 {
+[WUXC_VERSION_PREVIEW]
 [webhosthidden]
-unsealed runtimeclass RadioButtonsListView : Windows.UI.Xaml.Controls.ListView
+[default_interface]
+unsealed runtimeclass RadioButtonsElementFactory : MU_XC_NAMESPACE.ElementFactory
 {
-    RadioButtonsListView();
+    RadioButtonsElementFactory();
 }
 
+[WUXC_VERSION_PREVIEW]
 [webhosthidden]
 [default_interface]
-unsealed runtimeclass RadioButtonsListViewItem : Windows.UI.Xaml.Controls.ListViewItem
+unsealed runtimeclass ColumnMajorUniformToLargestGridLayout : MU_XC_NAMESPACE.NonVirtualizingLayout
```

Comment:
From the PR my understanding was this was internal, do we think it's necessary to be public? Can we just create and use it from code-behind for now?

Reply:
In the PR I'm setting the repeater's layout in the RadioButtons template. We could move this to code behind and not expose this type, but we'd be preventing the consumer from setting their own layout by retemplating. (unless we can use the facny new ReadLocalValue api I just learned about to only set the layout if the consumer hasn't already set one).

Reply:
The other aspect to this, which hasn't been updated in the xaml file cause I didn't give kiki an new one yet (sorry!) is that we are setting the layouts column and row spacing properties to theme resources in the controls resource dictionary. This would mean we'd have to do these theme resource look ups from code behind i think and that's been problematic in the past.

Reply:
I see, then having it be public in primitives seems fine.